BASE/src/Microsoft.ApplicationInsights/Channel/ITelemetryChannel.cs

```diff
@@ -31,5 +32,11 @@ public interface ITelemetryChannel : IDisposable
         /// Flushes the in-memory buffer.
         /// </summary>
         void Flush();
+
+        /// <summary>
+        /// Flushes the in-memory buffer asynchronously.
+        /// </summary>
+        /// <returns>The task to await.</returns>
```

Comment:
What is the meaning of the bool value? Does it indicate whether the Flush was successful? If it returns false, is there something else that the caller should do?